### DISCUSSION TITLE
[Idea] [Pricing] : Depending or we are Tax included or Tax ecluded results expected

### DISCUSSION THREAD

**Discussion by RosaBenouamer** [PrestaShop/PrestaShop#34262] [Ideas]
### The problem

Today depending on you are B to B (tax with VAT = no) or B to C (tax with VAT =yes) the result can be calculated in different ways.

### The solution

What is for you the good way among this one 👍 
<html>
<body>
<google-sheets-html-origin><style type="text/css"></style>

Solution 1 (Currently in Prestashop) : 0 digits after the decimal point | 0 |   |   |   |   |   |  
-- | -- | -- | -- | -- | -- | -- | --
Price with VAT | VAT | Unit Price | Count | Total | Total incl VAT display | Total excl VAT display | included VAT
Y | 21 | 1290 | 25 | 32250 | 32250.00 | **26650** |**5600.00**
N | 21 | 1,066.00 | 25 | 26650 | 32250.00 | 26650 | 5600.00
  |   |   |   |   |   |   |  
  |   |   |   |   |   |   |  
Solution 2 : 0 digits after the decimal point | 0 |   |   |   |   |   |  
Price with VAT | VAT | Unit Price | Count | Total | Total incl VAT display | Total excl VAT display | included VAT
Y | 21 | 1290 | 25 | 32247 | 32247.00 |**26650** |**5597.00**
N | 21 | 1,066.00 | 25 | 26650 | 32247.00 | 26650 | 5597.00
  |   |   |   |   |   |   |  
  |   |   |   |   |   |   |  
  |   |   |   |   |   |   |  
  |   |   |   |   |   |   |  
Solution 3 : 0 digits after the decimal point | 0 |   |   |   |   |   |  
Price with VAT | VAT |   | Count | Total | Total incl VAT display | Total excl VAT display | included VAT
Y | 21 | 1290 | 25 | 32250 | 32250.00 |**26653** | **5597.00**
N | 21 | 1,066.00 | 25 | 26650 | 32247.00 | 26650 | 5597.00
  |   |   |   |   |   |   |  
  |   |   |   |   |   |   |  
  |   |   |   |   |   |**Not displayed** |  

</body>
</html>

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
Hello @RosaBenouamer 

Could you translate what's available in the "Solution X some description"?

**Comment by RosaBenouamer** [comment]
Yes @kpodemski done :-)

**Comment by ChillCode** [comment]
Hi

What about when B2B have to apply two taxes? When we round using total, the taxes are stored with four decimals instead of two, the code doesn't apply the rounding in that case.

For example:

Tax 1 = 1.1036 in db = 1.10 rounded to two in the invoice tax tab.
Tax 2 = 0.1543 in db = 0.15 rounded to two in the invoice tax tab.

Total 1.25

But in the total taxes will reflect 1.26

Tax 1 + Tax 1 = 1.1036 + 0.1543 = rounded 1.26.

After solving almost all issues with taxes I found that almost all errors comes because PS is calculating taxes again and again when is not necessary at all.

**Comment by RosaBenouamer** [comment]
Hello @ChillCode,
Your question make sens.
 I am currently discussing the subject [here](https://github.com/PrestaShop/PrestaShop/issues/33912#issuecomment-1748782474) and on slack on FOP.
It seems that accountable are not ok on the subject does we use sum of rounding group by tax or rounding of sum of taxes.
I will open another discussion on this subject which is a little bit diffferent of the current :-)
https://github.com/PrestaShop/PrestaShop/discussions/34327

**Comment by RosaBenouamer** [comment]
Thank you @cruftex for you availability for our meeting.
We have exchange about this subject and have corrected the solution 3 which seems to be the good one.

Here the three solutions corrected  :
Solution 1 (Currently in Prestashop) : 2  digits after the decimal point

| Price with VAT | VAT | Unit Price | Count | Total | Total incl VAT display | Total excl VAT display | included VAT |
|----------------|-----|------------|-------|-------|------------------------|------------------------|--------------|
| Y              | 21  | 1289.86    | 25    | 32246.50 | 32246.50 | 26650.00 | 5596.50     |
| N              | 21  | 1066.00    | 25    | 26650.00 | 32246.50 | 26650.00 | 5596.50     |




Solution 2 :  2 digits after the decimal point

| Price with VAT | VAT | Unit Price | Count | Total | Total incl VAT display | Total excl VAT display | included VAT |
|----------------|-----|------------|-------|-------|------------------------|------------------------|--------------|
| Y              | 21  | 1290.27    | 25    | 32256.79 | 32256.79 | 26658.50 | 5598.29     |
| N              | 21  | 1066.34    | 25    | 26658.50 | 32256.79 | 26658.50 | 5598.29     |




Solution 3 :  2 digits after the decimal point

| Price with VAT | VAT |    | Count | Total | Total incl VAT display | Total excl VAT display | included VAT |
|----------------|-----|----|-------|-------|------------------------|------------------------|--------------|
| Y              | 21  | 1290.28 | 25 | 32257.00 | 32257.00 | 26658.68 | 5598.32     |
| N              | 21  | 1066.34 | 25 | 26658.60 | 32256.91 | 26658.60 | 5598.306     |


[Untitled spreadsheet.xlsx](https://github.com/PrestaShop/PrestaShop/files/13587445/Untitled.spreadsheet.xlsx)
